### PR TITLE
Refactor to use generic pub/sub from rclcpp

### LINKED ISSUE #28: Use generic pub/sub from rclcpp
Now that implementation for generic publishers and subscriptions exists in rclcpp, we can remove the duplicate code from domain_bridge.

### PR DESCRIPTION
**Pull request by rebecca-butler**
Closes #28

### CODE REVIEW

**Comment by rebecca-butler** [comment]
Paging @ivanpauno for review.

File: src/domain_bridge/domain_bridge.cpp [hunk 1]
```diff
@@ -107,41 +108,33 @@ class DomainBridgeImpl
...
-    const rosidl_message_type_support_t & typesupport_handle,
     rclcpp::CallbackGroup::SharedPtr group)
   {
-    auto publisher = std::make_shared<GenericPublisher>(
-      node->get_node_base_interface().get(),
-      typesupport_handle,
-      topic_name,
-      qos);
+    auto publisher = node->create_generic_publisher(topic_name, type, qos);
     node->get_node_topics_interface()->add_publisher(publisher, std::move(group));
```
Comment (ivanpauno) [inline src/domain_bridge/domain_bridge.cpp:119]: you can delete this line, `create_generic_publisher()` already does this [here](https://github.com/ros2/rclcpp/blob/1037822a63330495dcf0de5e8f20544375a5f116/rclcpp/include/rclcpp/create_generic_publisher.hpp#L63).

IIRC, you can pass a callback group in the [options](https://github.com/ros2/rclcpp/blob/1037822a63330495dcf0de5e8f20544375a5f116/rclcpp/include/rclcpp/node.hpp#L290) argument.

File: src/domain_bridge/domain_bridge.cpp [hunk 2]
```diff
@@ -107,41 +108,33 @@ class DomainBridgeImpl
...
+      type,
       qos,
       [publisher](std::shared_ptr<rclcpp::SerializedMessage> msg) {
         // Publish message into the other domain
-        auto serialized_data_ptr = std::make_shared<rcl_serialized_message_t>(
-          msg->get_rcl_serialized_message());
-        publisher->publish(serialized_data_ptr);
+        publisher->publish(*msg);
       });
     node->get_node_topics_interface()->add_subscription(subscription, std::move(group));
```
Comment (ivanpauno) [inline src/domain_bridge/domain_bridge.cpp:139]: same here

File: src/domain_bridge/domain_bridge.cpp [hunk 3]
```diff
@@ -50,7 +50,8 @@ class DomainBridgeImpl
   using NodeMap = std::unordered_map<std::size_t, std::shared_ptr<rclcpp::Node>>;
   using TopicBridgeMap = std::map<
     TopicBridge,
-    std::pair<std::shared_ptr<GenericPublisher>, std::shared_ptr<GenericSubscription>>>;
+    std::pair<std::shared_ptr<rclcpp::GenericPublisher>,
+    std::shared_ptr<rclcpp::GenericSubscription>>>;
```
[the remark this answers is not in the document]
Comment (rebecca-butler) [reply src/domain_bridge/domain_bridge.cpp]: Fixed.

File: src/domain_bridge/domain_bridge.cpp (hunk 1 above)
Comment (rebecca-butler) [reply src/domain_bridge/domain_bridge.cpp]: I've removed this line and passed the callbacks into the options argument.

**Review by ivanpauno — APPROVED** [review]
LGTM!
I don't have write access, so waiting for @jacobperron to merge it.

P.S.: Maybe we can be given write access here (?)

**Comment by jacobperron** [comment]
@clalancette Can you give the ROS 2 team write access to this repo? thanks!

**Pull request merged**